File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int KthSmallest(TreeNode root, int k) {
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode current = root;
        int count = 0;

        while (current != null || stack.Count > 0) {
            while (current != null) {
                stack.Push(current);
                current = current.left;
            }
            
            current = stack.Pop();
            count++;
            
            if (count == k) {
                return current.val;
            }
            
            current = current.right;
        }
        
        // In case of an invalid input (k out of range), though constraints ensure k is always valid.
        throw new ArgumentException("k is out of range");
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
    let count=0, result=null;
    const inOrder=(node)=>{
        if(!node ||result !==null) return;
        inOrder(node.left);
        count++;
        if(count===k){result=node.val; return;}
        inOrder(node.right);
    };
    inOrder(root);
    return result;
};